modules/tracking/src/trackerKCF.cpp

```diff
@@ -626,36 +709,32 @@ namespace cv{
   /*
    * calculate the detection response
    */
-  void TrackerKCFImpl::calcResponse(const Mat _alphaf, const Mat _k, Mat & _response) const {
+  void TrackerKCFImpl::calcResponse(const Mat _alphaf, const Mat _kf, Mat & _response, Mat & _spec) const {
     //alpha f--> 2channels ; k --> 1 channel;
-    Mat _kf;
-    fft2(_k,_kf);
-    Mat spec;
-    mulSpectrums(_alphaf,_kf,spec,0,false);
+    mulSpectrums(_alphaf,_kf,_spec,0,false);
     ifft2(spec,_response);
   }
 
   /*
    * calculate the detection response for splitted form
    */
-  void TrackerKCFImpl::calcResponse(const Mat _alphaf, const Mat _alphaf_den, const Mat _k, Mat & _response) const {
-    Mat _kf;
-    fft2(_k,_kf);
-    Mat spec;
-    Mat spec2=Mat_<Vec2d >(_k.rows, _k.cols);
-    std::complex<double> temp;
-
-    mulSpectrums(_alphaf,_kf,spec,0,false);
-
-    for(int i=0;i<_k.rows;i++){
-      for(int j=0;j<_k.cols;j++){
-        temp=std::complex<double>(spec.at<Vec2d>(i,j)[0],spec.at<Vec2d>(i,j)[1])/(std::complex<double>(_alphaf_den.at<Vec2d>(i,j)[0],_alphaf_den.at<Vec2d>(i,j)[1])/*+std::complex<double>(0.0000000001,0.0000000001)*/);
-        spec2.at<Vec2d >(i,j)[0]=temp.real();
-        spec2.at<Vec2d >(i,j)[1]=temp.imag();
+  void TrackerKCFImpl::calcResponse(const Mat _alphaf, const Mat _alphaf_den, const Mat _kf, Mat & _response, Mat & _spec, Mat & _spec2) const {
+
+    mulSpectrums(_alphaf,_kf,_spec,0,false);
+
+    //z=(a+bi)/(c+di)=[(ac+bd)+i(bc-ad)]/(c^2+d^2)
+    double den;
```

Comment:
Just for the sake of good practice, initialise the variable upon declaration (e.g. to zero).